### PR TITLE
Key Value Store

### PR DESCRIPTION
**Pull request by prashantchhabra89**
Summary - Implement Key Value Store
1. dictionary.py uses dictionary of python to store key value to store key and value
2. I have 3 static methods(get, put and delete) which can be called directly using class-name.method name
3. I have made sure that only one instance of class is created
4. dictionary object is converted into json string and written to file
5. used dictionaryTest.py to test dictionary file

Question - We can make synchronized object but do we really want that??

### CODE REVIEW

File: dictionary.py [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+        return dictionary.instantiation().getInternal(k)
+    
+    def getInternal(self,k):
+        return dictionary.dictObj.get(k)
+    
+    def writeToFile(self):
+        #conversion of dictionary object to json string
+        jsonstring=json.dumps(dictionary.dictObj)
+        #just clearing file before writing anything to it
+        open('file.txt', 'w').close()
```
Comment (sheoranjs24) [inline dictionary.py]: Change name of the file. It is very generic name.

File: dictionary.py [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+    
+    flagForObjInstantiation=0
+    
+    classObj = {}
+    
+    #This method is called from various static methods to make sure that this class has
+    #only one object
+    @staticmethod
+    def instantiation():
+        if dictionary.flagForObjInstantiation == 0:
```
Comment (sheoranjs24) [inline dictionary.py:18]: Make sure indentation is correct. And that you are using spaces (not tabs).

File: dictionary.py [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+        #todo call internal fn
+
+    @staticmethod
+    def get(k):
+        return dictionary.instantiation().getInternal(k)
+    
+    def getInternal(self,k):
+        return dictionary.dictObj.get(k)
+    
+    def writeToFile(self):
```
Comment (sheoranjs24) [inline dictionary.py:56]: We may want to make if more efficient, so that we are not writing the whole file after each change. Maybe in the future.

Comment (sheoranjs24) [inline dictionary.py:56]: Make sure the method names matches with the api code.

**Comment by sheoranjs24** [comment]
I don't think we need synchronization at this point of time.

**Comment by prashantchhabra89** [comment]
Changes made according to comments. Started new pull request hence, closing this one

**Pull request closed without merging**